src/vimscript/pattern.ts

```diff
@@ -83,8 +84,20 @@ export class Pattern {
       fromPosition = args.fromPosition;
     }
 
-    const haystack = vimState.document.getText();
-    const startOffset = vimState.document.offsetAt(fromPosition);
+    let haystack: string;
+    let searchOffset: number;
+    if (this.inSelection && vimState.lastVisualSelection) {
+      // TODO: This is not exactly how Vim implements in-selection search (\V%), see :help \%V for more info.
+      haystack = vimState.document.getText(
+        new Range(vimState.lastVisualSelection.start, vimState.lastVisualSelection.end)
+      );
+      searchOffset = vimState.document.offsetAt(vimState.lastVisualSelection.start);
+    } else {
+      haystack = vimState.document.getText();
+      searchOffset = 0;
```

Comment:
Something seems a bit buggy when I test it out...

Shouldn't this be `searchOffset = vimState.document.offsetAt(fromPosition)`, and just use that instead of `searchOffset`?

Reply:
Hmm, can you describe what seems buggy? maybe I didn't test this use case.

I'll check it again, but the searchOffset is required since the regex match will return the index relative to the selection start, and fromPosition is just where to start searching.

Reply:
It doesn't always include every match within the visual selection; I think it's starting from the wrong position

Reply:
I think it fixed now, by 1a96e70.